Draft: [IMP] open_academy: New menu item T#59063

EmilioSerna opened the pull request:
Added new entries (courses) as demo data in a new menu item view

EmilioSerna committed:
[IMP] open_academy: New menu item T#59063

Added new entries as demo in a new menu item view

EmilioSerna commented:
Please review @andreaarce

andreaarce commented:
Your commit needs to be more descriptive, you add a menu item, but for what?, where?

andreaarce commented:
@frahikLV Could you help me with the review of this MR LGTM but what do you think?

andreaarce commented:
@EmilioSerna  Please remove the draft label

frahikLV commented on open_academy/__manifest__.py:
```diff
@@ -14,4 +14,13 @@
+
+    "data": [
+        "security/ir.model.access.csv",
+        "views/course.xml",
```
```suggestion
        "views/course_views.xml",
```

I hate this suggestion 🤣, but it's OCA guideline https://github.com/OCA/odoo-community.org/blob/master/website/Contribution/CONTRIBUTING.rst#id10

frahikLV commented on open_academy/__manifest__.py:
```diff
@@ -14,4 +14,13 @@
+    ],
+
+    "demo": [
+        "demo/course.xml",
```
```suggestion
        "demo/course_demo.xml",
```

frahikLV commented on open_academy/views/course.xml:
```diff
@@ -1,11 +1,10 @@
-    <menuitem id="open_academy_root_menu" name="Courses" sequence="1"/>
-    <menuitem id="course_menu" name="Courses" sequence="1"
-    action="course_action" parent="open_academy_root_menu"/>
+    <menuitem id="course_menu_form" name="Courses" sequence="1"
+    action="course_action_list"/>
```
Why?

EmilioSerna replied:
That's what I understood in the OCA guideline, is it better to use a root menu?

frahikLV replied:
You had the format
root_menu_app
 |
 |- Menu courses

Now, you are assigning the action to the root menu, eliminating the possibility of adding more menus in the future (which you will have to do), please revert this change.

frahikLV commented on open_academy/views/course.xml:
```diff
@@ -1,11 +1,10 @@
 <?xml version="1.0" encoding="utf-8"?>
 <odoo>
-    <record id="action_list_ideas" model="ir.actions.act_window">
+    <record id="course_action_list" model="ir.actions.act_window">
```
```suggestion
    <record id="course_action" model="ir.actions.act_window">
```

frahikLV reviewed (changes requested):
@EmilioSerna

EmilioSerna committed:
[FIX] open_academy: Fixes module installation T#59063

The property _name of Course cannot be open_academy.course because the tag res_model in views/course.xml cannot parse open_academy.course, simply renaming _name as just "course" fixes this issue.

frahikLV commented on open_academy/models/course.py:
```diff
@@ -2,7 +2,7 @@
 class Course(models.Model):
-    _name = "open_academy.course"
+    _name = "course"
 
```
missing `_description`

EmilioSerna commented:
I did all the suggested changes, please review 🙂️ @andreaarce @frahikLV

andreaarce commented:
@frahikLV please review

frahikLV reviewed (approved):
@andreaarce LGTM
